scripts/check_e2e_tests_are_captured_in_ci_test.py

```diff
@@ -0,0 +1,241 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/check_e2e_tests_are_captured_in_ci.py."""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+
+from core.tests import test_utils
+import python_utils
+import utils
+
+from . import check_e2e_tests_are_captured_in_ci
+
+DUMMY_TEST_SUITES = ['oneword', 'twoWords']
+
+DUMMY_CONF_FILES = os.path.join(
+    os.getcwd(), 'core', 'tests', 'data', 'dummy_ci_tests')
+
+
+class CheckE2eTestsCapturedInCI(test_utils.GenericTestBase):
+    """Test the methods which performs travis.yml and
+    protractor.conf.js sync checks.
+    """
+
+    def _mock_read_protractor_conf_file(self):
+        protractor_config_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, 'dummy_protractor.conf.js'), 'r').read()
+        return protractor_config_file
+
+    def _mock_read_travis_yml_file(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, '.dummy_travis.yml'), 'r').read()
+        travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        return travis_ci_dict
+
+    def _mock_read_protractor_conf_file_fail(self):
+        protractor_config_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, 'protractor_has_extra_test.conf.js'),
+                'r').read()
+        return protractor_config_file
+
+    def _mock_read_travis_yml_file_fail(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, '.travis_has_extra_test.yml'), 'r').read()
+        travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        return travis_ci_dict
+
+    def _return_empty_list(self):
+        return []
+
+    def test_read_protractor_file(self):
+        expected_protractor_config_file = python_utils.open_file(
+            os.path.join(
+                os.getcwd(), 'core', 'tests', 'protractor.conf.js'), 'r').read()
+        actual_protractor_config_file = (
+            check_e2e_tests_are_captured_in_ci.read_protractor_conf_file())
+        self.assertEqual(expected_protractor_config_file,
+                         actual_protractor_config_file)
+
+    def test_read_travis_ci_file(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(os.getcwd(), '.travis.yml'), 'r').read()
+        expected_travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
```

Comment:
Actually this should be hardcoded in the test.

Reply:
I don't quite understand, should the hardcode the whole Travis ci dict?

Reply:
Done